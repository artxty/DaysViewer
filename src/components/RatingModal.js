import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import { Modalize } from 'react-native-modalize';
import iosColors from 'ios-colors';
import {
  Text,
  View,
  StyleSheet,
  TextInput,
  Button,
} from 'react-native';

import { getDateFromDateId, format } from '../helpers/date';
import RatingPicker from './RatingPicker';

const styles = StyleSheet.create({
  header: {
    paddingVertical: 15,
    fontSize: 24,
    fontWeight: '600',
    textAlign: 'center',
  },
  container: {
    paddingHorizontal: 15,
  },
  textInput: {
    fontSize: 16,
    fontWeight: '500',
    marginVertical: 15,
    minHeight: 50,
    padding: 15,
    borderRadius: 15,
    backgroundColor: iosColors.systemGray6.light,
  },
  footer: {
    marginBottom: 10,
  },
});

const RatingModal = React.forwardRef(({ dayData, onSave }, ref) => {
  const [title, setTitle] = useState();
  const [rating, setRating] = useState();
  const [note, setNote] = useState();

  useEffect(() => {
    const dayDate = getDateFromDateId(dayData.id);
    setTitle(format(dayDate, 'PPP'));
    setRating(dayData.rating);
    setNote(dayData.note);
  }, [dayData.id]);

  const handleSavePress = () => onSave({
    id: dayData.id,
    rating: rating || 0,
    note: note || '',
  });

  return (
    <Modalize
      ref={ref}
      modalTopOffset={50}
      adjustToContentHeight
      HeaderComponent={() => <Text style={styles.header}>{title}</Text>}
      FooterComponent={() => <View style={styles.footer}><Button title="Done" onPress={handleSavePress} /></View>}
    >
      <View style={styles.container}>
        <RatingPicker
          rating={rating}
          onPress={setRating}
          containerStyle={styles.ratingPicker}
        />
        <TextInput
          multiline
          placeholder="Note"
          maxLength={400}
          numberOfLines={1}
          onChangeText={setNote}
          value={note}
          style={styles.textInput}
        />
      </View>
    </Modalize>
  );
});

RatingModal.propTypes = {
  dayData: PropTypes.exact({
    id: PropTypes.string.isRequired,
    rating: PropTypes.number.isRequired,
    note: PropTypes.string.isRequired,
  }).isRequired,
  onSave: PropTypes.func.isRequired,
};

export default RatingModal;
