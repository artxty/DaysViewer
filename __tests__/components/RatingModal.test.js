import React from 'react';
import renderer from 'react-test-renderer';

import RatingModal from '../../src/components/RatingModal';

jest.mock('react-native', () => {
  const reactNative = jest.requireActual('react-native');
  return {
    __esModule: true,
    View: 'View',
    Text: 'Text',
    Button: 'Button',
    TextInput: 'TextInput',
    StyleSheet: reactNative.StyleSheet,
  };
});

jest.mock('react-native-modalize', () => ({ Modalize: 'Modalize' }));
jest.mock('../../src/components/RatingPicker', () => 'RatingPicker');

describe('RatingModal', () => {
  const ref = React.createRef(null);

  it('renders correctly', () => {
    const dayData = {
      id: '2020-04-01',
      rating: 4,
      note: 'My little note right here!',
    };

    const tree = renderer.create(
      <RatingModal
        ref={ref}
        onSave={jest.fn()}
        dayData={dayData}
      />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
