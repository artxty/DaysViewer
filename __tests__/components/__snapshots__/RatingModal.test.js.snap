// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RatingModal renders correctly 1`] = `
<Modalize
  FooterComponent={[Function]}
  HeaderComponent={[Function]}
  adjustToContentHeight={true}
  modalTopOffset={50}
>
  <View
    style={
      Object {
        "paddingHorizontal": 15,
      }
    }
  >
    <RatingPicker
      onPress={[Function]}
    />
    <TextInput
      maxLength={400}
      multiline={true}
      numberOfLines={1}
      onChangeText={[Function]}
      placeholder="Note"
      style={
        Object {
          "backgroundColor": "rgb(242, 242, 247)",
          "borderRadius": 15,
          "fontSize": 16,
          "fontWeight": "500",
          "marginVertical": 15,
          "minHeight": 50,
          "padding": 15,
        }
      }
    />
  </View>
</Modalize>
`;
